### ISSUE TITLE
[LinkField] Add ability to pick LinkField links form content items with Autoroute 

### ISSUE THREAD

**Issue by xperiandri** [OrchardCMS/OrchardCore#11191]
### Is your feature request related to a problem? Please describe.
It is a huge ache to sync links within content items with Autoroute links to other content items.

### Describe the solution you'd like
Ability to enable content item selection within Autoroute part like it is done in Content Menu Item.

### Describe alternatives you've considered
Create another part that can do that.

**Comment by xperiandri** [comment]
The issue is that you must extract it from that content item yourself. And your cannot set an arbitrary link of your need

**Comment by MikeKry** [comment]
I guess this will need further description with examples.

**Comment by sebastienros** [comment]
> And your cannot set an arbitrary link if your need

You want a content item link, but sometime not? You will need to create your own custom field.

**Comment by xperiandri** [comment]
Yes:
1. Usually, you want to reference some content item.
2. Sometimes you want to reference a fragment on that content item link
3. Sometimes you want a custom link to another website
4. Sometimes you want to open it in a new tab or a new window

**Comment by MikeKry** [comment]
I have done something similiar by creating custom field with dropdown selector that shows content picker or link field based on selection in dropdown, it is just an extra layer over existing fields (new custom field - drivers, views and models.) Got that working for headless / graphql, not sure how it would perform in other recipes.

But It would be obviously better to have one field that is dedicated to this functionality (or ideally a text field that with possibility to select editor display of different kinds - content picker, link field, user picker...)

**Comment by xperiandri** [comment]
@MikeMike is you code available somewhere?

**Comment by mikemike** [comment]
Think you've tagged the wrong Mike here :)

**Comment by MikeKry** [comment]
Not really, also I have modified it overtime that it depends on many other things, but I could try to get some basic example if I will have some time.

All you need is to create new field then you can get inspiration from existing fields from OC how to create view and drivers.
Simplest case is to copy views from Link field & Content Picker field, create new view that will show link or content picker and create driver that merges behavior from both of them. A bit challenging might be creating settings for that field, because each field has different settings.


But @sebastienros added this to backlog, so I guess it means he is planning to create a new field for that in OC?
